app/models/manageiq/providers/openstack/manager_mixin.rb

```diff
@@ -314,16 +309,27 @@ def verify_amqp_credentials(_options = {})
   end
   private :verify_amqp_credentials
 
-  def verify_credentials(auth_type = nil, options = {})
-    auth_type ||= 'default'
-
-    raise MiqException::MiqHostError, "No credentials defined" if self.missing_credentials?(auth_type)
+  def verify_event_monitor(_options = {})
+    require 'manageiq/providers/openstack/legacy/openstack_event_monitor'
+    OpenstackEventMonitor.available?(event_monitor_options)
+  rescue => err
+    miq_exception = translate_exception(err)
+    raise miq_exception || err
+  end
 
-    options[:auth_type] = auth_type
-    case auth_type.to_s
-    when 'default' then verify_api_credentials(options)
-    when 'amqp' then    verify_amqp_credentials(options)
-    else;           raise "Invalid OpenStack Authentication Type: #{auth_type.inspect}"
+  def verify_credentials(auth_type = nil, options = {})
+    case auth_type
+    when :default
+      verify_api_credentials
+    when :amqp
+      verify_amqp_credentials
+    else
+      verify_api_credentials
+
+      capabilities["events"] = !!event_monitor_available?
+      save! if changed?
+
+      true
```

Comment:
I don't love how this looks, but the general idea is if you request a specific auth type we'll do only that, but if auth_type is nil (which is the case when doing the scheduled or from the UI authentication check) we'll check API and event_monitor.